### PR TITLE
librbd: optimize header file dependency & modify related file.  

### PR DESCRIPTION
**Pull request by cxwshawn**
same work as https://github.com/ceph/ceph/pull/9336, https://github.com/ceph/ceph/pull/9307, https://github.com/ceph/ceph/pull/9287

 Signed-off-by: Xiaowei Chen <chen.xiaowei@h3c.com>

### CODE REVIEW

**Comment by dillaman** [comment]
retest this please

**Comment by dillaman** [comment]
Lots of commented out header files -- if they aren't needed, remove them

File: src/common/event_socket.h [hunk 1]
```diff
@@ -52,9 +53,7 @@ class EventSocket {
           ret = -errno;
         else
           ret = 0;
-        break;
```
Comment (dillaman) [inline src/common/event_socket.h]: Why did you remove the preprocessor directives and case breaks?

**Comment by cxwshawn** [comment]
@dillaman  sorry for the no-update event_socket.h modification, have fixed, @dillaman @liewegas please help review.

**Comment by dillaman** [comment]
retest this please

**Comment by dillaman** [comment]
@cxwshawn rebase is required

**Comment by cxwshawn** [comment]
@dillaman sorry, I don't get it, u mean rebase into one commit or  ?

**Comment by dillaman** [comment]
@cxwshawn it cannot be merged in current form (see the status at the bottom of the PR). it needs to be rebased against the current state of the master branch to fix the merge conflicts.

**Comment by cxwshawn** [comment]
@dillaman Ah ~ oh , rebased, please help review again !

File: src/librbd/Utils.h [hunk 2]
```diff
@@ -7,8 +7,7 @@
 #include "include/rados/librados.hpp"
 #include "include/Context.h"
 #include <type_traits>
-
-class Context;
+#include "common/WorkQueue.h"
```
Comment (dillaman) [inline src/librbd/Utils.h]: This will pull ```WorkQueue``` into scope for lots of classes -- some which don't require it.

File: src/librbd/image/RefreshParentRequest.h [hunk 3]
```diff
@@ -4,7 +4,6 @@
 #ifndef CEPH_LIBRBD_IMAGE_REFRESH_PARENT_REQUEST_H
 #define CEPH_LIBRBD_IMAGE_REFRESH_PARENT_REQUEST_H
 
-#include "include/int_types.h"
```
Comment (dillaman) [inline src/librbd/image/RefreshParentRequest.h]: You've yanked this from a lot of files -- where are the int types being defined?

Comment (cxwshawn) [inline src/librbd/image/RefreshParentRequest.h]: @dillaman since as we know include/int_types.h is so basic header file, and on librbd level(image parent level), there is librbd/parent_types.h, and this has included int_types.h, so there is really no need to reinclude.

Comment (dillaman) [inline src/librbd/image/RefreshParentRequest.h]: @cxwshawn I personally don't like that logic since you are introducing new inter-dependencies instead of just cleaning out unnecessary includes.

Comment (cxwshawn) [inline src/librbd/image/RefreshParentRequest.h]: @dillaman hmm,  how to say, just like "include/types.h", if we have include this header file, is there any need to reinclude
#include <string>
#include <list>
#include <set>
#include <map>
#include <vector>
#include <iostream>
#include <iomanip>
?
the purpose of design include/types.h header file is to decrease reinclude these files again, under some circumstances, I think this is essential.

Comment (dillaman) [inline src/librbd/image/RefreshParentRequest.h]: "include/types.h" pulls in lots of unneeded things ... and it appears the goal of this PR is to remove these unneeded includes.  However, I'd say removing includes because you want to rely on other header files (which aren't designed to be catch-all) makes it more difficult to understand how header dependencies are being satisfied. 

TL;DR: I love all the cleanup of unused header files but I really don't like implicitly relying on other random header files (a la parent_types.h) to satisfy dependencies.

**Comment by cxwshawn** [comment]
@dillaman  fixed again, please review, thanks.

**Comment by cxwshawn** [comment]
@dillaman PING!

**Comment by dillaman** [comment]
lgtm
